internal/broker/poller.go

```diff
@@ -43,3 +43,18 @@ func (p *PassthroughPoller) Invoke(logic func() (bool, error)) error {
 	}
 	return err
 }
+
+type TimerPoller struct {
+	PollInterval time.Duration
+	PollTimeout  time.Duration
+	Log func(args ...any)
+}
+
+func (p *TimerPoller) Invoke(logic func() (bool, error)) error {
+	var start = time.Now()
+	result := wait.PollImmediate(p.PollInterval, p.PollTimeout, func() (bool, error) {
+		return logic()
+	})
+	p.Log(fmt.Sprintf("Waiting for the status took: %v", time.Since(start)))
```

Comment:
It is not easy to comprehend that we call `logic` to get `status`.  Either comment or more consistent names would be beneficial.

Reply:
Changed the message.